cagov/cannabis.ca.gov
# Task 269/end to end tests

This branch adds end to end tests using https://playwright.dev/

The tests run on push (currently on this branch) controlled by the e2e.yml file which runs the build and kicks off the tests

Tests cover:
- About page load, verify visible h1
- Page passes axe accessibility test with no failures
- Click a content navigation link
- Verify the Google Analytics event count stored in dataLayer has incremented to report the anchor link click
- Find the per page feedback element on the page
- Verify per page feedback textarea is not visible
- Click the positive feedback button
- Verify per page feedback textarea is now visible as it should be after the initial button was clicked

Tests fire up their own dev server and kill it after running. This works but could use a better way of doing this.

I made 2 HTML changes to pass the axe a11y check: adding header element around the header section and aside around the sidebar. This was recommended by the "include all page elements in landmarks" rule

The change in the scripts section of package.json reorder scripts alphabetically and adds:
- ```test``` to run these e2e tests
- ```test:server``` script to launch a static site server. This is used in the beforeAll hook inside the test script file
- ```test:setup``` script to install playwright tools needed inside git action
